device/core/devdoc/device_method/method_client_requirements.md

```diff
@@ -0,0 +1,47 @@
+# DirectMethodClient requirements
+
+The `DirectMethodClient` class is in charge of making edgeHub method API calls. It is a private class that shall be used by the `ModuleClient` to make method calls and the SDK user should only use the `ModuleClient` object.
+
+## Overview
+```typescript
+export class DirectMethodClient {
+  constructor(authProvider: AuthenticationProvider);
+  invokeMethod(deviceId: string, moduleId: string, methodParams: DirectMethodParams, callback: DirectMethodCallback): void;
+  setOptions(options: any): void;
+  updateSharedAccessSignature(sharedAccessSignature: string, callback: (err?: Error) => void): void;
+}
+```
+
+## API
+*Note: this API is internal only and is used solely by the `ModuleClient` object.*
+
+### invokeMethod(deviceId: string, moduleId: string, methodParams: DirectMethodParams, callback: DirectMethodCallback): void
+
+**SRS_NODE_DEVICE_METHOD_CLIENT_16_006: [** The `invokeMethod` method shall get the latest credentials by calling `getDeviceCredentials` on the `AuthenticationProvider` object. **]**
+
+**SRS_NODE_DEVICE_METHOD_CLIENT_16_007: [** The `invokeMethod` method shall create a `RestApiClient` object if it does not exist and update its shared access signature. **]**
```

Comment:
>The `invokeMethod` method shall create a `RestApiClient` object if it does not exist and update its shared access signature [](start = 44, length = 123)

This is a vaguely worded requirement because it's not clear if it always updates the SAS, or only if it creates the new RestApiClient object.  At the very least, this requirement describes two flows that probably deserves 2 test cases.  I needed to read the code to makes sure it did what I thought.  I recommend having 2 requirements: 1) It creates the RestApiClient if it doesn't exist (with uses a valid SAS), or 2) if the RestApiClient already exists, it updates the SAS.